packages/stark-ui/src/modules/restrict-input-directive/directives/restrict-input.directive.spec.ts

```diff
@@ -107,14 +152,72 @@ describe("RestrictInputDirective", () => {
 		it("should NOT prevent any of the values given in the configuration from being typed in the input", () => {
 			const inputElement: DebugElement = fixture.debugElement.query(By.css("input"));
 
-			let keyPressEventDefaultPrevented: boolean = triggerKeyPressEvent(inputElement, "1");
+			let keyPressEventDefaultPrevented: boolean = triggerKeyPressEvent(inputElement, "9");
 			expect(keyPressEventDefaultPrevented).toBe(false);
 
-			keyPressEventDefaultPrevented = triggerKeyPressEvent(inputElement, "9");
+			keyPressEventDefaultPrevented = triggerKeyPressEvent(inputElement, "1");
 			expect(keyPressEventDefaultPrevented).toBe(false);
 
 			keyPressEventDefaultPrevented = triggerKeyPressEvent(inputElement, "0");
 			expect(keyPressEventDefaultPrevented).toBe(false);
 		});
+
+		it("should prevent any of the values given in the configuration from being pasted in the input", () => {
+			expect(fixture).toBeDefined();
+			const inputElement: DebugElement = fixture.debugElement.query(By.css("input"));
+
+			let pastEventDefaultPrevented: boolean = triggerPasteEvent(inputElement, "a");
+			expect(pastEventDefaultPrevented).toBe(true);
+
+			pastEventDefaultPrevented = triggerPasteEvent(inputElement, "B");
+			expect(pastEventDefaultPrevented).toBe(true);
+
+			pastEventDefaultPrevented = triggerPasteEvent(inputElement, "-");
+			expect(pastEventDefaultPrevented).toBe(true);
+		});
+
+		// tslint:disable-next-line:no-identical-functions
+		it("should NOT prevent any of the values given in the configuration from being pasted in the input", () => {
+			expect(fixture).toBeDefined();
+			const inputElement: DebugElement = fixture.debugElement.query(By.css("input"));
+
+			let pastEventDefaultPrevented: boolean = triggerPasteEvent(inputElement, "9");
+			expect(pastEventDefaultPrevented).toBe(false);
+
+			pastEventDefaultPrevented = triggerPasteEvent(inputElement, "1");
+			expect(pastEventDefaultPrevented).toBe(false);
+
+			pastEventDefaultPrevented = triggerPasteEvent(inputElement, "0");
+			expect(pastEventDefaultPrevented).toBe(false);
+		});
+
+		it("should prevent any of the values given in the configuration from being drop in the input", () => {
+			expect(fixture).toBeDefined();
+			const inputElement: DebugElement = fixture.debugElement.query(By.css("input"));
+
+			let dropEventDefaultPrevented: boolean = triggerDropEvent(inputElement, "a");
+			expect(dropEventDefaultPrevented).toBe(true);
+
+			dropEventDefaultPrevented = triggerDropEvent(inputElement, "B");
+			expect(dropEventDefaultPrevented).toBe(true);
+
+			dropEventDefaultPrevented = triggerDropEvent(inputElement, "-");
+			expect(dropEventDefaultPrevented).toBe(true);
+		});
+
+		// tslint:disable-next-line:no-identical-functions
```

Comment:
In order to avoid repeating ourselves, maybe could you add assert functions for these events ? 
For instance, you could have something like this:

```typescript
function assertDropEventsValidity(inputElement: DebugElement, values: string[], shouldBeValid: boolean): void {
	for (const value of values) {
		const dropEventDefaultPrevented = triggerDropEvent(inputElement, value);
		expect(dropEventDefaultPrevented).toBe(shouldBeValid);
	}
}

// ...

it("should prevent any of the values given in the configuration from being drop in the input", () => {
	expect(fixture).toBeDefined();
	const inputElement: DebugElement = fixture.debugElement.query(By.css("input"));

	assertDropEventsValidity(inputElement, ["9", "0", "1"], false);
});
```

And apply the same for the other events. What do you think about it? 😊